From dbf058395a8b8a6cc477c1a77adfa14ff1556047 Mon Sep 17 00:00:00 2001
From: Satheesh Kumar Pallemoni <spallemo@codeaurora.org>
Date: Wed, 8 Apr 2020 12:36:07 +0530
Subject: [PATCH 11/12] A2dpSrc: DEV-UI support for Blutooth Audio codec.

Due to enhancements in the DEV-UI by AOSP, Blutooth Audio
codec menu showing nothing in developer settings. Due to
length mismatch of the bluetooth_a2dp_codec_titles and
bluetooth_a2dp_codec_summaries in settings leading to
this issue.

Issue has been fixed by introducing QVA codecs too, so
that length won't mismatch.

CRs-Fixed: 2624640
Change-Id: I849e340152be5043782fa8274636a534c147c8bb
---
 res/layout/bluetooth_audio_codec_dialog.xml            |  8 ++++++++
 .../AbstractBluetoothDialogPreferenceController.java   |  5 ++++-
 .../bluetooth/BluetoothCodecDialogPreference.java      | 10 ++++++++++
 3 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/res/layout/bluetooth_audio_codec_dialog.xml b/res/layout/bluetooth_audio_codec_dialog.xml
index 9636427055..1fd9de8021 100644
--- a/res/layout/bluetooth_audio_codec_dialog.xml
+++ b/res/layout/bluetooth_audio_codec_dialog.xml
@@ -51,9 +51,17 @@
                 android:id="@+id/bluetooth_audio_codec_aptx_hd"
                 layout="@layout/preference_widget_dialog_radiobutton"/>
 
+            <include
+                android:id="@+id/bluetooth_audio_codec_aptx_adaptive"
+                layout="@layout/preference_widget_dialog_radiobutton"/>
+
             <include
                 android:id="@+id/bluetooth_audio_codec_ldac"
                 layout="@layout/preference_widget_dialog_radiobutton"/>
+
+            <include
+                android:id="@+id/bluetooth_audio_codec_aptx_twsp"
+                layout="@layout/preference_widget_dialog_radiobutton"/>
         </RadioGroup>
 
         <include
diff --git a/src/com/android/settings/development/bluetooth/AbstractBluetoothDialogPreferenceController.java b/src/com/android/settings/development/bluetooth/AbstractBluetoothDialogPreferenceController.java
index d66b8d8dc8..c0143ba82a 100644
--- a/src/com/android/settings/development/bluetooth/AbstractBluetoothDialogPreferenceController.java
+++ b/src/com/android/settings/development/bluetooth/AbstractBluetoothDialogPreferenceController.java
@@ -38,7 +38,9 @@ public abstract class AbstractBluetoothDialogPreferenceController extends
 
     private static final String TAG = "AbstractBtDlgCtr";
 
-    protected static final int[] CODEC_TYPES = {BluetoothCodecConfig.SOURCE_CODEC_TYPE_LDAC,
+    protected static final int[] CODEC_TYPES = {BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_TWSP,
+            BluetoothCodecConfig.SOURCE_CODEC_TYPE_LDAC,
+            BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_ADAPTIVE,
             BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_HD,
             BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX,
             BluetoothCodecConfig.SOURCE_CODEC_TYPE_AAC,
@@ -224,6 +226,7 @@ public abstract class AbstractBluetoothDialogPreferenceController extends
             Log.d(TAG, "Unable to get highest codec. Configs are empty");
             return BluetoothCodecConfig.SOURCE_CODEC_TYPE_INVALID;
         }
+        Log.d(TAG, "CODEC_TYPES len: " + CODEC_TYPES.length + " codec_config len: " + configs.length);
         for (int i = 0; i < CODEC_TYPES.length; i++) {
             for (int j = 0; j < configs.length; j++) {
                 if ((configs[j].getCodecType() == CODEC_TYPES[i])) {
diff --git a/src/com/android/settings/development/bluetooth/BluetoothCodecDialogPreference.java b/src/com/android/settings/development/bluetooth/BluetoothCodecDialogPreference.java
index 3f1d23cf5b..7b1f38613d 100644
--- a/src/com/android/settings/development/bluetooth/BluetoothCodecDialogPreference.java
+++ b/src/com/android/settings/development/bluetooth/BluetoothCodecDialogPreference.java
@@ -22,12 +22,16 @@ import android.widget.RadioGroup;
 
 import com.android.settings.R;
 
+import android.util.Log;
+
 /**
  * Dialog preference to set the Bluetooth A2DP config of codec
  */
 public class BluetoothCodecDialogPreference extends BaseBluetoothDialogPreference implements
         RadioGroup.OnCheckedChangeListener {
 
+    private static final String TAG = "BtCodecDlgPref";
+
     public BluetoothCodecDialogPreference(Context context) {
         super(context);
         initialize(context);
@@ -60,13 +64,19 @@ public class BluetoothCodecDialogPreference extends BaseBluetoothDialogPreferenc
         mRadioButtonIds.add(R.id.bluetooth_audio_codec_aac);
         mRadioButtonIds.add(R.id.bluetooth_audio_codec_aptx);
         mRadioButtonIds.add(R.id.bluetooth_audio_codec_aptx_hd);
+        mRadioButtonIds.add(R.id.bluetooth_audio_codec_aptx_adaptive);
         mRadioButtonIds.add(R.id.bluetooth_audio_codec_ldac);
+        mRadioButtonIds.add(R.id.bluetooth_audio_codec_aptx_twsp);
         String[] stringArray = context.getResources().getStringArray(
                 R.array.bluetooth_a2dp_codec_titles_cm);
+
+        Log.e(TAG, "a2dp_codec_titles array length: " + stringArray.length);
         for (int i = 0; i < stringArray.length; i++) {
             mRadioButtonStrings.add(stringArray[i]);
         }
+
         stringArray = context.getResources().getStringArray(R.array.bluetooth_a2dp_codec_summaries_cm);
+        Log.e(TAG, "a2dp_codec_summaries array length: " + stringArray.length);
         for (int i = 0; i < stringArray.length; i++) {
             mSummaryStrings.add(stringArray[i]);
         }
-- 
2.28.0

